cluster-state-service/handler/store/instancestore.go

```diff
@@ -130,19 +134,51 @@ func (instanceStore eventInstanceStore) ListContainerInstances() ([]types.Contai
 }
 
 // FilterContainerInstances returns all container instances from the datastore that match the provided filters
-func (instanceStore eventInstanceStore) FilterContainerInstances(filterKey string, filterValue string) ([]types.ContainerInstance, error) {
-	if len(filterKey) == 0 || len(filterValue) == 0 {
-		return nil, errors.New("Filter key and value cannot be empty")
+func (instanceStore eventInstanceStore) FilterContainerInstances(filterMap map[string]string) ([]types.ContainerInstance, error) {
+	if len(filterMap) == 0 {
+		return nil, errors.New("There has to be at least one filter")
+	}
+
+	filters := make([]string, 0, len(filterMap))
+	for k := range filterMap {
+		filters = append(filters, k)
+	}
+
+	if !instanceStore.areFiltersValid(filters) {
+		return nil, errors.Errorf("At least one of the provided filters '%v' is not supported.", filters)
+	}
+
+	for _, f := range filters {
```

Comment:
Nope, don't need both. Using filterMap is probably cleaner although it may not buy us anything else. Will change.